Server/mods/deathmatch/logic/luadefs/CLuaResourceDefs.cpp

```diff
@@ -1476,24 +1476,28 @@ bool CLuaResourceDefs::isResourceProtected(CResource* const resource)
 std::vector<std::string> CLuaResourceDefs::GetResourceFiles(lua_State* luaVM, std::optional<CResourceFile::eResourceCategory> type,
                                                             std::optional<CResource*> resource) noexcept
 {
+    using eResourceCategory = CResourceFile::eResourceCategory;
+
     if (!type)
-        type = CResourceFile::eResourceCategory::ALL;
+        type = eResourceCategory::ALL;
 
     if (!resource)
         resource = &lua_getownerresource(luaVM);
 
     const auto resourceFiles = (*resource)->GetResourceFiles();
 
-    std::vector<std::string> files;
-    files.reserve(resourceFiles.size());
+    std::unordered_set<std::string> files;
     for (const auto& file : resourceFiles)
     {
-        if (file->GetResourceCategoryType() == type.value()
-            || type.value() == CResourceFile::eResourceCategory::ALL)
-        {
-            files.push_back(file->GetName());
-        }
+        if (file->GetResourceCategoryType() != type.value() && type.value() != eResourceCategory::ALL)
+            continue;
+        files.insert(file->GetName());
```

Comment:
Previous varian was more readable. Why did you change it?

Reply:
previous variant was basically the same.
This leaves room for more conditions to come.

Reply:
There is no 'early continue' rule. You obfuscate the code with overusing 'continue' in loops. And you don't really need to use early return for small code blocks.
You did the code less readable and maintainable.
```c++
if (file->GetResourceCategoryType() == type.value() || type.value() == eResourceCategory::ALL)
    files.insert(file->GetName());
```
Is fine condition, that describes the function logic.

Reply:
There is early-return which is basically the same but whatever.
I'd rather go with continue rather than indentation

Reply:
Your `continue` creates indentation too. Please, keep it simple

> There is early-return which is basically the same but whatever.

Not really. Multiple 'continue' conditions can make a loop harder for reading.
There is no goal to use minimal indentation. The code should primarily describe the logic.
You're overusing 'early return'

Reply:
Don't resolve conversations without changes silently

Reply:
Im not going to change that. We didnt continue the discussion for some time, hence I resolved it.
Conditions may change. Better to overuse early return than to not use it at all/not use it often enough.
Result? Mta code with huge if blocks.

Reply:
> We didnt continue the discussion for some time, hence I resolved it.

It looks like you tried to hide requested changes. Be civil. Maybe another mta member will points that I'm wrong or confirms my position.

> Conditions may change.

This is not how overused code looks like. It's better to keep it readable for future changes.

> Better to overuse early return than to not use it at all/not use it often enough.

You deal in absolutes. Overused principles are harmful for readability and complexity. Use them only when they improve your code. In this example you trade readability for principles.

Reply:
> > We didnt continue the discussion for some time, hence I resolved it.
> 
> It looks like you tried to hide requested changes. Be civil. Maybe another mta member will points that I'm wrong or confirms my position.
> 
> > Conditions may change.
> 
> This is not how overused code looks like. It's better to keep it readable for future changes.
> 
> > Better to overuse early return than to not use it at all/not use it often enough.
> 
> You deal in absolutes. Overused principles are harmful for readability and complexity. Use them only when they improve your code. In this example you trade readability for principles.

If more contributors are opposed against this change, I will reconsider

Reply:
Again?

Reply:
> Again?

No one was interested in this convo for 2 weeks, give it a break.
No contributors were opposed against this so Im leaving it as it is.

Reply:
I also see the early return overuse here.

Reply:
> No contributors were opposed against this so Im leaving it as it is.

And nobody confirm your position. Don't hide requested changes. Stale conversations should be dismissed by another  members.

Reply:
> And nobody confirm your position

Im following the guidelines is'all.

Reply:
you interpreter these guidelines very differently